### PR TITLE
Make tracing work for fee currency txs

### LINKED ISSUE #69: Check and correct tracing after fee currency addition
Part of [celo-org/optimism#6 ](https://github.com/celo-org/op-geth/issues/130)

- ~~One test was removed in [d694fb7defc907ad269d0a179c1d2168c2fabfbd](https://github.com/celo-org/op-geth/pull/28/commits/d694fb7defc907ad269d0a179c1d2168c2fabfbd) , because of a bug in the test setup. This also exists in upstream geth, and should be reported there.~~ Test is fixed in a proper way in https://github.com/celo-org/op-geth/commit/c66ca8bf7a8c63ae54e44f4566e206cd1a4fa204#diff-13c76087ba15a5cdb7f3003416a918dd7269b3fac232fc98908d8be927fed6d9.
- See also https://github.com/celo-org/celo-blockchain-planning/issues/66

### PR DESCRIPTION
**Pull request by karlb**
Closes https://github.com/celo-org/op-geth/issues/69

Added tests for the following tracers:
* Call tracer: same result as when paying in native token (debit/credit ignored)
* Prestate tracer: includes debit/credit, since the point of the prestate tracer is to have everything you need to execute the tx and we would miss out on the necessary fee currency code and state otherwise
* Prestate tracer diff mode: includes debit/credit => the gas payment is visible in the diff, as it is for native token

### CODE REVIEW

File: eth/tracers/native/prestate.go [hunk 1]
```diff
@@ -93,6 +93,8 @@ func newPrestateTracer(ctx *tracers.Context, cfg json.RawMessage) (*tracers.Trac
 			OnTxStart: t.OnTxStart,
 			OnTxEnd:   t.OnTxEnd,
 			OnOpcode:  t.OnOpcode,
+			// Celo
+			TraceDebitCredit: true,
```
Comment (palango) [inline eth/tracers/native/prestate.go:97]: Maybe add an explicit `TraceDebitCredit: false` to the call tracer as well?

File: eth/tracers/native/prestate.go [hunk 2]
```diff
@@ -158,6 +160,9 @@ func (t *prestateTracer) OnTxStart(env *tracing.VMContext, tx *types.Transaction
 	t.lookupAccount(from)
 	t.lookupAccount(t.to)
 	t.lookupAccount(env.Coinbase)
```
Comment (palango) [inline eth/tracers/native/prestate.go:162]: When this is added here, should we add accounts which receive fees here as well, like the `FeeHandler`?

**Review by palango — APPROVED** [review]
Looks great, awesome that we can use the test infrastructure.

Left two questions, but feel free to ignore them.

File: eth/tracers/native/prestate.go (hunk 2 above)
Comment (karlb) [reply eth/tracers/native/prestate.go:162]: I only thought about the fee currency txs in this PR, where the `FeeHandler` is part of the fee currency's storage. But for a normal tx and enables Cel2, that might be necessary. I'll take a look at that case.

**Comment by karlb** [comment]
Todo: should the FeeHandler address show up in the diff mode tracing?

EDIT:  tipReceiver and baseFeeReceiver are the same in the test, therefore only a single address (plus tx sender) is touched.

File: eth/tracers/native/prestate.go (hunk 2 above)
Comment (karlb) [reply eth/tracers/native/prestate.go:162]: Followup issue is at https://github.com/celo-org/op-geth/issues/216.

**Pull request merged**